File: src/test/java/net/tassia/event/EventHandlerTest.java
package net.tassia.event;

import org.junit.Test;

public class EventHandlerTest {

	private boolean test = false;

	@Test
	public void test() {
		EventManager manager = new EventManager();
		manager.registerEvent(TestEvent.class);

		manager.registerListeners(this);

		manager.callEvent(new TestEvent());

		assert test;
	}

	@EventHandler
	public void onEvent(TestEvent event) {
		test = true;
	}



	private static class TestEvent extends Event {
	}

}

File: src/test/kotlin/net/tassia/event/EventHandlerTest.kt
package net.tassia.event

import kotlin.test.Test

class EventHandlerTest {

	class TestEvent : Event()

	private var test1 = false

	@Test
	fun test() {
		val manager = EventManager.newDefault()
		manager.registerEvent(TestEvent::class)

		manager.registerListeners(this)

		manager.callEvent(TestEvent())

		assert(test1)
	}

	@EventHandler
	fun onEvent(event: TestEvent) {
		test1 = true
	}

}
